--- a/gamemodes/zcity/gamemode/modes/hl2dm/cl_hl2dm.lua
+++ b/gamemodes/zcity/gamemode/modes/hl2dm/cl_hl2dm.lua
@@ -253,8 +253,12 @@
 		end
 
 		function but:DoClick()
-			if ply:IsBot() then chat.AddText(Color(255,0,0), "no, you can't") return end
-			gui.OpenURL("https://steamcommunity.com/profiles/"..ply:SteamID64())
+			local id64 = ply:SteamID64()
+			if ply:IsBot() then
+				id64 = ply:GetNWString("zcSteamID64", "")
+				if id64 == "" then return end
+			end
+			gui.OpenURL("https://steamcommunity.com/profiles/"..id64)
 		end
 
 		DScrollPanel:AddItem(but)
