--- a/lua/weapons/homigrad_base/sh_ammo.lua
+++ b/lua/weapons/homigrad_base/sh_ammo.lua
@@ -13,7 +13,9 @@
 			Mass = 10,
 			AirResistMul = 0.0001,
 			NumBullet = 1,
+			Ballistics = {construction = "fmj", expand = nil, fragChance = nil, fragMax = nil, boneMul = nil},
 		}
+construction legend: fmj, ap, sp, jhp, frangible, shot, flechette, slug, blunt
 */
 
 function SWEP:ApplyAmmoChanges(type_)
